### PR TITLE
chore: upgrade dependencies to latest version

### PR DESCRIPTION
**Pull request by vannyle**
This pull request brings the dependencies upgrade to the latest version. 

**Main major updates**
- react@17 => react@18.2.0
- react-dom@17 => react-dom@18.2.0
- @mdx-js/react@1 => @mdx-js/react@2.1.5
- gatsby-plugin-mdx@3 => gatsby-plugin-mdx@4.3.0

**Details**
`gatsby-plugin-mdx@^4` is a complete rewrite of the original plugin with the goal of making the plugin faster, and compatible with [MDX v2](https://mdxjs.com/blog/v2/). See the migration guide from v3 to v4 for [more](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#migrating-from-v3-to-v4).

NOTES: MDX v2 does NOT support the autolink anymore as in the v1. All links in markdown files should be listed using the syntax `[Link](URL)` instead of `<https://github.com>`. Learn more: https://github.com/mdx-js/mdx/issues/1049

### CODE REVIEW

File: gatsby/create-pages.js [hunk 1]
```diff
@@ -2,7 +2,6 @@ const createBlogPages = require('./create-blog-pages');
 const createBlogPosts = require('./create-blog-posts');
 const createDocPages = require('./create-doc-pages');
 const createReleaseNotePost = require('./create-release-note-post');
-
 // const createStaticPages = require('./create-static-pages');
```
Comment (sashasushko) [inline gatsby/create-pages.js:5]: Can we remove this line?

Comment (sashasushko) [reply gatsby/create-pages.js:5]: Or add ToDo why it's commented

File: src/components/pages/release-notes/release-note-list/release-note-list.jsx [hunk 2]
```diff
@@ -1,16 +1,37 @@
+import MarkdownIt from 'markdown-it';
 import PropTypes from 'prop-types';
 import React from 'react';
 
 import Aside from 'components/pages/release-notes/aside';
-import Content from 'components/shared/content';
+
+const Content = ({ content }) => {
+  const md = new MarkdownIt({
```
Comment (sashasushko) [inline src/components/pages/release-notes/release-note-list/release-note-list.jsx:8]: Can we move it outside `<Content/>`?

File: src/components/pages/release-notes/release-note-list/release-note-list.jsx [hunk 3]
```diff
@@ -1,16 +1,37 @@
...
+    typographer: true,
+  });
+
+  const result = md.render(content);
+
+  return (
+    <div
+      className="prose-lg prose md:prose-base xs:prose-code:break-words"
+      dangerouslySetInnerHTML={{ __html: result }}
+    />
```
Comment (sashasushko) [inline src/components/pages/release-notes/release-note-list/release-note-list.jsx:20]:
```suggestion
  return (
    <div
      className="prose-lg prose md:prose-base xs:prose-code:break-words"
      dangerouslySetInnerHTML={{ __html: md.render(content) }}
    />
```

**Review by sashasushko — APPROVED** [review]

**Review by bolotskydev — APPROVED** [review]

**Pull request merged**